helios-services/src/main/java/com/spotify/helios/agent/TaskMonitor.java

```diff
@@ -242,6 +243,20 @@ private void updateState(final TaskStatus.State state) {
   }
 
   /**
+   * Propagate the container error by setting it and committing the update.
+   */
+  private void updateContainerError(final String containerError) {
+    statusUpdater.setContainerError(containerError);
+    // Commit and push a new status
+    try {
+      statusUpdater.update();
+    } catch (InterruptedException e) {
+      // TODO: propagate interrupt instead?
+      Thread.currentThread().interrupt();
```

Comment:
Yeah - this comment was ganked from the same `updateState` method I ganked most of it from. I could get rid of both?

As an aside, I'm also open to implementations that avoid me having to call `update()` the `statusUpdater` twice.